tgui/packages/tgui/interfaces/Biogenerator.tsx

```diff
@@ -202,44 +212,84 @@ const ItemList = (props) => {
       disabled,
       max_amount,
       amount,
-      setAmount,
     };
   });
-  return items.map((item) => (
-    <Table.Row key={item.id}>
-      <Table.Cell>
-        <span
-          className={classes(['design32x32', item.id])}
-          style={{
-            verticalAlign: 'middle',
-          }}
-        />{' '}
-        <b>{item.name}</b>
-      </Table.Cell>
-      <Table.Cell collapsing>
-        <NumberInput
-          value={item.amount}
-          step={1}
-          width="35px"
-          minValue={1}
-          maxValue={item.max_amount}
-          onChange={(value) => item.setAmount(value)}
-        />
-      </Table.Cell>
-      <Table.Cell collapsing>
-        <Button
-          fluid
-          align="right"
-          content={parseFloat((item.cost * item.amount).toFixed(2)) + ' BIO'}
-          disabled={item.disabled}
-          onClick={() =>
-            act('create', {
-              id: item.id,
-              amount: item.amount,
-            })
-          }
-        />
-      </Table.Cell>
-    </Table.Row>
-  ));
+
+  const [itemAmounts, setItemIDAmount] = useState(() => {
+    return items.map((item) => {
+      return {
+        id: item.id,
+        amount: item.amount,
+      };
+    });
+  });
+
+  const setItemAmount = (itemID: number, newAmount: number) => {
+    setItemIDAmount((oldState) => {
```

Comment:
usestate should not be made this complex

Reply:
If we make a component for each item then we will have like 10 states for 10 items which is what u wanted to avoid.

Yes the value is complex but having 1 state with a large value is better for performance(memory vice) than having 10+ states each with a simple value?.

I don't know, i want your thoughts on this

Reply:
I'll still make the change though, no worries

Reply:
I did this in #83589. it's significantly simpler this way

Reply:
this would have never threw a bluescreen had they not been (effectively) nesting components inside of another component `itemslist`. i dont think the answer is making one blanket state for the entire itemslist

Reply:
I'll trust your PR since it does everything. Thanks for the insights.

Hope you're doing well :)

Reply:
You too!